securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/service/decision/FilePaymentConsentDecisionService.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright © 2020 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.service.decision;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.forgerock.securebanking.openbanking.uk.rcs.client.idm.PaymentConsentService;
+import com.forgerock.securebanking.openbanking.uk.rcs.client.idm.dto.consent.FRFilePaymentConsent;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FilePaymentConsentDecisionService extends PaymentConsentDecisionService {
+
+    public FilePaymentConsentDecisionService(PaymentConsentService paymentConsentService,
+                                             ObjectMapper objectMapper,
+                                             PaymentConsentDecisionUpdater paymentConsentDecisionUpdater) {
+        super(paymentConsentService, objectMapper, paymentConsentDecisionUpdater);
+    }
+
+    @Override
+    protected Class<FRFilePaymentConsent> getConsentClass() {
+        return FRFilePaymentConsent.class;
```

Comment:
Instead of duplicating the code in each of these `ConsentDecisionService` classes, the approach I'm using means there's virtually nothing to test for each payment type 👍🏻